### ISSUE TITLE
slight.exe doesn't run on WS2019 or WS2022 due to missing dlls

### ISSUE THREAD

**Issue by jsturtevant** [deislabs/spiderlightning#378]
**Description of the bug**
When running on Windows Server 2019/2022 slight doesn't run:

```
PS C:\Program Files\containerd\release> .\slight.exe -h
<no output>
```

**To Reproduce**
Run either WS2022 and WS2019

```
cmd /c ver
Microsoft Windows [Version 10.0.20348.1607]
```

install:
```
curl.exe -LO https://github.com/deislabs/spiderlightning/releases/download/v0.4.1/slight-windows-x86_64.tar.gz
tar.exe xvf .\slight-windows-x86_64.tar.gz
x release/slight.exe
```

![image](https://user-images.githubusercontent.com/648372/231516893-c5b1396e-9baf-4215-a984-2f3a1226b77d.png)

**Additional context**

We may want to link the runtime for windows statically as many of the binaries. info in https://stackoverflow.com/a/44387312/697126

**Comment by jsturtevant** [comment]
adding a `.cargo/config.toml` with 

```
[target.x86_64-pc-windows-msvc]
rustflags = ["-C", "target-feature=+crt-static"]
```

seems to have fixed the issue with the `vsruntime.dll` loading but still getting ssl related library loading issues:

![image](https://user-images.githubusercontent.com/648372/231522197-234781de-ba5b-48fb-9822-88f19a6b2afd.png)

**Comment by jsturtevant** [comment]
after running `choco install openssl -y` I am able to run it. We should look into alternatives for how openssl is used.

**Comment by Mossaka** [comment]
@jsturtevant I am not sure what's the best practices for how openssl should be used. Any suggestions? 

I should point out that our Makefile has a [command](https://github.com/deislabs/spiderlightning/blob/main/Makefile#L60-L70) for install all deps on Windoes. Shall we make this more visiable for people trying to compile slight on their host?

**Comment by jsturtevant** [comment]
it's not just compiling.  If I try to install the [released binary ](https://github.com/deislabs/spiderlightning/releases/download/v0.4.1/slight-windows-x86_64.tar.gz) on a fresh machine, you will need this additional install of `openssl` which IMO isn't a great expereince.





> I am not sure what's the best practices for how openssl should be used. Any suggestions?

I am not sure, I will need to look more deeply into how it is being used.

**Comment by D1esel-Dev** [comment]
Yes!! i just had an issue where there was no output aswell, now i fixed it because all i did was read this issue and installed openssl with choco: 
>PS C:\Users\zer0> choco install openssl -y
Chocolatey v2.1.0
Installing the following packages:
openssl
By installing, you accept licenses for the packages.
Progress: Downloading openssl 1.1.1.2100... 100%
openssl v1.1.1.2100 [Approved]
openssl package files install completed. Performing other installation steps.
Installing 64-bit openssl...
openssl has been installed.
WARNING: No registry key found based on  'OpenSSL-Win'
PATH environment variable does not have C:\Program Files\OpenSSL-Win64\bin in it. Adding...
WARNING: OPENSSL_CONF has been set to C:\Program Files\OpenSSL-Win64\bin\openssl.cfg
  openssl can be automatically uninstalled.
Environment Vars (like PATH) have changed. Close/reopen your shell to
 see the changes (or in powershell/cmd.exe just type `refreshenv`).
 The install of openssl was successful.
  Software installed to 'C:\Program Files\OpenSSL-Win64\'

Chocolatey installed 1/1 packages.
 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).
PS C:\Users\zer0> slight -?
error: unexpected argument '-?' found

Usage: slight.exe [OPTIONS] <COMMAND>

For more information, try '--help'.
PS C:\Users\zer0> slight -h
Usage: slight.exe [OPTIONS] <COMMAND>

Commands:
  run      Run slight providing a config and a module
  secret   Add a secret to the application
  add      Download a SpiderLightning interface
  new      Start a new Slight project
  buildjs  Build a JS Slight project
  help     Print this message or the help of the given subcommand(s)

Options:
  -c, --config <CONFIG>
  -h, --help             Print help
  -V, --version          Print version
PS C:\Users\zer0> slight new -n spidey@v0.5.1 rust && cd spidey
PS C:\Users\zer0\spidey>
